test/spec_set_x_forwarded_proto_header.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require_relative 'helper'
+
+
+describe Rack::SetXForwardedProtoHeader do
+  response = lambda {|e| [200, {}, []]  }
```

Comment:
I would switch to:

```ruby
  response = lambda {|e| [200, {}, [e['HTTP_X_FORWARDED_PROTO']]]  }
```

Then you can more easily test expected behavior by just checking for the result in the body.

Reply:
I'm unclear on why this makes things easier to test.

From a readability point of view, as the code stands in my PR, I can see a benefit in having tests that show that we're asserting that the header value has changed. If we were to return that header value in the body, we would have tests that assert some value of the body, abstracting the intention by one step.

But, open to feedback, this is your project.

Reply:
One advantage of returning the value in the body is it allows for a more functional approach.  With your current approach, you are relying on a side effect of changing `env`.  However, it's probably fine to leave as is.